proxy.js

```diff
@@ -24,12 +24,28 @@
   let lastRevokeFn = null;
 
   /**
+   * @param {*} o
+   * @return {boolean} whether this is probably an Object
+   */
+  function isObject(o) {
+    return o instanceof Object || (!!o && (typeof o == 'object' || typeof o == 'function'));
+  }
+
+  /**
+   * @param {*} o
+   * @return {boolean} whether this is probably a Function
+   */
+  function isFunction(o) {
+    return o instanceof Function || (!!o && typeof o == 'function');
+  }
+
+  /**
    * @constructor
    * @param {!Object} target
    * @param {{apply, construct, get, set}} handler
    */
   scope.Proxy = function(target, handler) {
-    if (!(target instanceof Object && handler instanceof Object)) {
+    if (!isObject(target) || !isObject(handler)) {
```

Comment:
maybe just split this condition into two conditions, and just use the typeof check in each, not need to introduce `isObject()` abstraction.